.github/workflows/mypy.yaml

```diff
@@ -0,0 +1,45 @@
+name: Mypy CI
+on:
+  push:
+    branches:
+      - develop
+      - master
+  pull_request:
+    branches:
+      - develop
+      - master
+
+concurrency:
+
+  group: "${{ github.ref }}-${{ github.head_ref }}"
```

Comment:
Please use the grouping which we use in gh-ci.yaml This one is prone to collisions